src/overlays/actors/ovl_En_Dnp/z_en_dnp.c

```diff
@@ -22,61 +65,454 @@ const ActorInit En_Dnp_InitVars = {
+    } else if (!(gSaveContext.weekEventReg[23] & 0x20)) {
+        if (Actor_HasParent(&this->actor, globalCtx)) {
+            func_8013AED4(&this->unk_322, 0, 7);
+            this->unk_322 &= ~0x500;
+            this->actor.parent = NULL;
+            this->unk_32E = 0;
+            this->actionFunc = func_80B3D338;
+            ret = true;
+        }
+    }
+    return ret;
+}
+
+s32 func_80B3D044(EnDnp* this, GlobalContext* globalCtx) {
+    s32 ret = false;
+
+    if (globalCtx->csCtx.state != 0) {
+        if (!(this->unk_322 & 0x200)) {
+            this->unk_322 |= (0x200 | 0x10);
+            this->actor.flags &= -2;
+            this->unk_324 = 0xFF;
+        }
+        func_8013AED4(&this->unk_322, 0, 7);
+        this->actionFunc = func_80B3D11C;
+        ret = true;
+    } else if (this->unk_322 & 0x200) {
+        this->actor.flags |= 1;
+        func_8013AED4(&this->unk_322, 3, 7);
+        this->unk_322 &= ~(0x200 | 0x10);
+        this->actionFunc = func_80B3D2D4;
+    }
+    return ret;
+}
+
+static s32 D_80B3DE58[] = {
+    0x00172000, 0x0D040005, 0x0E09670C, 0x100E0968, 0x0C100E09, 0x6F0C0F09, 0x700C1000,
+};
+
+void func_80B3D11C(EnDnp* this, GlobalContext* globalCtx) {
+    static s32 D_80B3DE74[] = {
+        0, 16, 14, 10, 18, 12, 5, 7, 2, 19, 21, 22, 24, 8,
+    };
+
+    u32 temp_v0;
+    s32 val;
+
+    if (!(gSaveContext.weekEventReg[29] & 0x40) && (globalCtx->sceneNum == SCENE_MITURIN) &&
+        (globalCtx->csCtx.unk_12 == 0)) {
+        this->unk_322 |= 0x20;
+        gSaveContext.weekEventReg[29] |= 0x40;
+    }
+
+    if (func_800EE29C(globalCtx, 0x65)) {
+        temp_v0 = func_800EE200(globalCtx, 0x65);
+        val = globalCtx->csCtx.npcActions[temp_v0]->unk0;
+        if (this->unk_324 != (u8)val) {
+            func_80B3CC38(this, D_80B3DE74[val]);
+            if (this->unk_340 == 16) {
+                this->unk_322 |= 8;
+            } else {
+                this->unk_322 &= ~0x8;
+            }
+
+            if (this->unk_340 == 19) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_VO_DHVO04);
+            }
+
+            if (this->unk_340 == 24) {
+                this->unk_322 &= ~0x80;
+                this->unk_336 = 3;
+                this->unk_334 = 0;
+            }
+        }
+
+        this->unk_324 = val;
+        if (((this->unk_340 == 10) || (this->unk_340 == 12) || (this->unk_340 == 5) || (this->unk_340 == 19) ||
+             (this->unk_340 == 22) || (this->unk_340 == 24)) &&
+            func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+            func_80B3CC38(this, this->unk_340 + 1);
+        }
+        func_800EDF24(&this->actor, globalCtx, temp_v0);
+    }
+}
+
+void func_80B3D2D4(EnDnp* this, GlobalContext* globalCtx) {
+    if (this->unk_322 & 0x20) {
+        Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 3, 10920);
+    } else {
+        Math_ApproachS(&this->actor.shape.rot.y, this->actor.world.rot.y, 3, 10920);
```

Comment:
10920 = 0x2AA8, and both of these values are used throughout the file. Not sure if it should be decimal or hex, but sticking to one would be good

Reply:
I don't know what's meant to be decimal and what's meant to be hex.

Reply:
Rule of thumb is everything stays in hex unless it's colours or timer values, IIRC. Angles we are keeping in hex for now, may do a mass-replace later. However, this is actually 60 degrees as a hex angle. We may do a macro for this conversion, but not for now.